buildtest/buildsystem/base.py

```diff
@@ -266,6 +267,36 @@ def __init__(self, name, recipe, buildspec=None, testdir=None):
                 % (self.type, self.recipe.get("type"))
             )
 
+        self.shell = self.recipe.get("shell")
+        # if 'shell' key not found in Buildspec, let's set it to BUILDTEST_SHELL and log a message
+        if not self.shell:
```

Comment:
Do you want to make self.shell a property instead? I can see this being reused.

Reply:
Can you describe with example how you plan to reuse the shell decorator. I have been getting up to speed on decorators but haven't found a use-case. 

I am thinking you want to invoke BuilderBase class directly if i understand. I think changing shell from ``bash`` --> ``sh`` or ``sh`` --> ``bash`` would be okay for most part, but there are some minor differences between the two shells that can cause an error depending on how ``run`` section is setup. I was able to find this link https://www.gnu.org/software/bash/manual/html_node/Major-Differences-From-The-Bourne-Shell.html outlining some of the differences between ``bash`` and ``sh`` 

I don't see a usecase where we can allow a shell to change from ``bash`` or ``sh`` --> ``python`` since that would cause run section to break pretty dramatically.

Reply:
All the times in the code we do some action like "self.recipe.get("shell") we would want to just reference self.shell. And I was thinking this morning, since there are now a bunch of different parameters around a shell, would it make sense to have it be a class? E.g.,

```Python
self.shell = Shell(self.recipe.get('shell', '/bin/bash'))
self.shell.opts
self.shell.shebang
```
etc.

Reply:
I have been thinking about the ``shell`` key this is specific to ``LocalExecutor`` behavior since we are running test locally. This wont be relevant when running test with ``SlurmExector``. It's true we still need to detect the shell for writing the test, get test extension,  but execution of test is depended on the executor type. We also have ``executor`` key https://github.com/buildtesters/buildtest/blob/devel/buildtest/buildsystem/schemas/script/script-v0.0.1.schema.json#L39  which we have not implemented during the build stage. Right now everything defaults to LocalExecutor but again this is out of scope for this PR but worth pointing out.